benchmark/utils/ArgParse.swift

```diff
@@ -12,67 +12,200 @@
 
 import Foundation
 
-public struct Arguments {
-  public var progName: String
-  public var positionalArgs: [String]
-  public var optionalArgsMap: [String : String]
-
-  init(_ pName: String, _ posArgs: [String], _ optArgsMap: [String : String]) {
-    progName = pName
-    positionalArgs = posArgs
-    optionalArgsMap = optArgsMap
+enum ArgumentError: Error {
+  case missingValue(String)
+  case invalidType(value: String, type: String, argument: String?)
+  case unsupportedArgument(String)
+}
+
+extension ArgumentError: CustomStringConvertible {
+  public var description: String {
+    switch self {
+    case let .missingValue(key):
+      return "missing value for '\(key)'"
+    case let .invalidType(value, type, argument):
+      return (argument == nil)
+        ? "'\(value)' is not a valid '\(type)'"
+        : "'\(value)' is not a valid '\(type)' for '\(argument!)'"
```

Comment:
I thought it was helpful user feedback. For example, when you use a wrong tag, the message will read: `error: 'bogus' is not a valid 'BenchmarkCategory'`. So you'll know what to look for. 

The type-checking parser for `BenchmarkCategory` (`func tags`) is called from a nested context within the attribute parsers for `tags` and `skip-tags`, where it doesn't have access to the argument name. The best it could do, would be to say `'bogus' is not valid value'`. 

I'm not sure that's the right trade-off here. I think it's worth a little bit of code complexity for much better QoL.